app/controllers/facility_notifications_controller.rb

```diff
@@ -43,9 +43,13 @@ def send_notifications
     end
 
     sender = NotificationSender.new(current_facility, params[:order_detail_ids])
+    order_details = OrderDetail.where_ids_in(params[:order_detail_ids])
 
     if sender.perform
       flash[:notice] = send_notification_success_message(sender)
+      order_details.each do |order_detail|
```

Comment:
Sorry if I confused you. I wanted to bring up the Oracle thing  so you were aware of it and this seemed like a good place to bring it up. I think we can avoid the issue entirely by using `sender.order_details.each` now that that method is public. It also avoids an extra lookup and loading of objects into memory.